client/src/__tests__/QuestionsAnswers.test.js

```diff
@@ -223,45 +226,169 @@ describe('Renders the QnA Component, GET Request Pulls Data', () => {
+    },
+  ];
+  const grabQuestionsMock = jest.fn();
+  test('QnAList to render', () => {
+    render(<QnAList quests={quests} product={{}} grabQuestions={grabQuestionsMock} />);
     const QnAListRender = screen.getByTestId('QnAList Test');
+    const questButtons = screen.getByTestId('question-buttons');
     expect(QnAListRender).toBeInTheDocument();
+    expect(questButtons).toBeInTheDocument();
+  });
+  test('Show More Answered Questions to disappear when clicked', () => {
+    const { queryByDisplayValue, queryByText } = render(<QnAList quests={quests} product={{}} grabQuestions={grabQuestionsMock} />);
+    const showMoreButton = queryByDisplayValue('Show More Answered Questions');
+    fireEvent.click(showMoreButton);
+    expect(queryByDisplayValue('Show More Answered Questions')).toBeNull();
+    expect(queryByText('Show All Questions')).toBeNull();
+  });
+});
+
+describe('The QnAListEntry Component', () => {
+  const quests = [
+    {
+      question_id: 329977,
+      question_body: 'Ipsam aperiam enim.',
+      question_helpfulness: 27,
+      reported: false,
+      answers: {
+        3082776: {
+          id: 3082776,
+          body: 'Eaque voluptate qui sed sed quo dolorem enim eos.',
+          date: '2020-11-13T00:00:00.000Z',
+          answerer_name: 'Violette.Boyer18',
+          helpfulness: 7,
+          photos: [
+            'https://images.unsplash.com/photo-1519396317879-83334cb422f8?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80',
+          ],
+        },
+      },
+    },
+    {
+      question_id: 329976,
+      question_body: 'Veniam voluptatem est et quidem nemo similique occaecati.',
+      question_helpfulness: 25,
+      reported: false,
+      answers: {
+        3082759: {
+          id: 3082759,
+          body: 'Veniam in ipsam id assumenda.',
+          date: '2020-10-05T00:00:00.000Z',
+          answerer_name: 'Adan35',
+          helpfulness: 2,
+          photos: [],
+        },
+      },
+    },
+    {
+      question_id: 329975,
+      question_body: 'When in Rome.',
+      question_helpfulness: 25,
+      reported: false,
+      answers: {
+        3082759: {
+          id: 3082759,
+          body: 'Veniam in ipsam id assumenda.',
+          date: '2020-10-05T00:00:00.000Z',
+          answerer_name: 'Adan35',
+          helpfulness: 2,
+          photos: [],
+        },
+      },
+    },
+  ];
+  const grabQuestionsMock = jest.fn();
+  test('QnAListEntry to render two questions', () => {
+    const numEntry = 2;
+
+    const { queryAllByTestId } = render(
+      <div>
+        {quests.slice(0, numEntry).map((quest) => (
+          <QnAListEntry
+            quest={quest}
+            key={quest.question_id}
+            product={{}}
+            grabQuestions={grabQuestionsMock}
+          />
+        ))}
+      </div>,
+    );
+
+    const questionComponents = queryAllByTestId('individual-question-test');
+    expect(questionComponents.length).toBe(numEntry);
```

Comment:
Jests:
describe('The QnA Component)
You have a test commented out for rendering 2 questions at the start.
I see that you added some mock data to this and mapped over them to check how many entries there are. But I also think the first way you were doing it was super close and may have been a bit simpler.
I'm not entirely sure but I think all you would have to do is first, render the QnA component again in this test block using your quests variable, then use screen.getAllByTestId to get an array. Then with that array you can use just use the .length check in your expect to be 2.